helper/resource/testing_new_import_state.go

```diff
@@ -231,16 +231,40 @@ func testStepNewImportState(ctx context.Context, t testing.T, helper *plugintest
 		if err := runPlanChecks(ctx, t, plan, step.ImportPlanChecks.PreApply); err != nil {
 			return err
 		}
-	} else {
-		err = runProviderCommand(ctx, t, func() error {
-			return importWd.Import(ctx, resourceName, importId)
-		}, importWd, providers)
-		if err != nil {
-			return err
+
+		{
+			if kind.resourceIdentity() {
+				err := runProviderCommandApply(ctx, t, wd, providers)
+				if err != nil {
+					return fmt.Errorf("applying plan with import config: %s", err)
+				}
+
+				newStateJSON, err := runProviderCommandGetStateJSON(ctx, t, wd, providers)
+				if err != nil {
+					return fmt.Errorf("getting state after applying plan with import config: %s", err)
+				}
+
+				newIdentityValues := identityValuesFromState(newStateJSON, resourceName)
+				if !cmp.Equal(priorIdentityValues, newIdentityValues) {
+					return fmt.Errorf("importing resource %s: expected identity values %v, got %v", resourceName, priorIdentityValues, newIdentityValues)
+				}
+			}
```

Comment:
I feel like I'm more leaning towards not applying and just waiting for TF core to give us the data during plan for comparison (it sounds like it should be included)

At least how it is ATM, core is validating that the data doesn't change during apply, and if they remove that logic, the SDKs should probably add it.

Reply:
Mostly because we can't fully trust providers to apply what they plan (since the testing framework also supports SDKv2 😆 ), so if we can avoid it, I think we should

Reply:
Sounds good to me. Since we have a working implementation in this PR, it should be no trouble to switch to planned values & expect tests to remain green.

Reply:
We talked about this offline and we're good with keeping apply with the caveat that we'll remove it before full release